### PR TITLE
Implement SingleElementVectorINTEL decoration

### PR DESCRIPTION
**Pull request by NikitaRudenkoIntel**
According to the spec
https://github.com/intel/llvm/pull/1612

### CODE REVIEW

File: test/transcoding/SPV_INTEL_vector_compute/decoration_single_element_vector.ll [hunk 1]
```diff
@@ -0,0 +1,39 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv --spirv-ext=+SPV_INTEL_vector_compute --spirv-allow-unknown-intrinsics
+; RUN: llvm-spirv -r %t.spv -o %t.bc
+; RUN: llvm-dis %t.bc -o %t.ll
+; RUN: FileCheck %s --input-file %t.ll  -check-prefix=LLVM
```
Comment (AlexeySachkov) [inline test/transcoding/SPV_INTEL_vector_compute/decoration_single_element_vector.ll:5]: Please add checks for SPIR-V representation of the attribute as well

Comment (NikitaRudenkoIntel) [reply test/transcoding/SPV_INTEL_vector_compute/decoration_single_element_vector.ll:7]: Done

File: lib/SPIRV/SPIRVWriter.cpp [hunk 2]
```diff
@@ -617,6 +617,10 @@ void LLVMToSPIRV::transVectorComputeMetadata(Function *F) {
     BF->addDecorate(DecorationSIMTCallINTEL, SIMTMode);
   }
 
+  if (Attrs.hasAttribute(AttributeList::ReturnIndex,
+                         kVCMetadata::VCSingleElementVector))
+    BF->addDecorate(DecorationSingleElementVectorINTEL);
```
Comment (AlexeySachkov) [inline lib/SPIRV/SPIRVWriter.cpp:622]: I've just realized that this implementation is not really conformant with the spec, as per extension spec this decoration should be applied to a type and not to function or its argument:

> Apply to a 'Scalar' type to indicate that the type represents a single-element vector type in the high-level language.

Comment (NikitaRudenkoIntel) [reply lib/SPIRV/SPIRVWriter.cpp:630]: Spec was updated, now the patch conforms with it

**Review by AlexeySachkov — APPROVED** [review]

**Pull request merged**